### ISSUE TITLE
Users Screen not loading

### ISSUE THREAD

**Issue by deanbruhn** [nilsteampassnet/TeamPass#4000]
### Steps to reproduce
1. Created a clone server running on http instead of https for upgrade testing
2. Upgraded from 2.1.27.36 to 3.1.0
3. Upgraded from 3.1.0 to 3.1.1.24
4. Both upgrades completed successfully, and after the upgrade from 2.1.27.36 to 3.1.0 the users panel worked fine
5. Now when I login as the admin user and click on Users from the left side menu, I getan error

### Expected behaviour
The users page should load so I can see and edit the users

### Actual behaviour
I get an error telling me that it's an Ajax error. 

### Server configuration
**Operating system**:
Debian 11

**Web server:**
Apache 2.4.56

**Database:**
MariaDB 10.7.8

**PHP version:**
8.2.14

**Teampass version:** 
3.1.1.24

**Teampass configuration file:**
                                                                                                                     
<?php
// DATABASE connexion parameters
define("DB_HOST", "<omitted");
define("DB_USER", ",<omitted>");
define("DB_PASSWD", "<omited>");
define("DB_NAME", "teampass");
define("DB_PREFIX", "teampass_");
define("DB_PORT", "3306");
define("DB_ENCODING", "utf8");
//define("DB_SSL", false); // if DB over SSL then comment this line
// if DB over SSL then uncomment the following lines
define("DB_SSL", array(
    "key" => "",
    "cert" => "",
    "ca_cert" => "",
    "ca_path" => "",
    "cipher" => ""
));
define("DB_CONNECT_OPTIONS", array(
    MYSQLI_OPT_CONNECT_TIMEOUT => 10
));
define("SECUREPATH", "/etc/teampass");
define("SECUREFILE", "teampass-seckey.txt");
define("IKEY", "");
define("SKEY", "");
define("HOST", "");

if (isset($_SESSION['settings']['timezone']) === true) {
    date_default_timezone_set($_SESSION['settings']['timezone']);
}

**Updated from an older Teampass or fresh install:**
PLEASE attach to this issue the file `/includes/config/tp.config.php`.

<?php
global $SETTINGS;
$SETTINGS = array (
    'max_latest_items' => '10',
    'enable_favourites' => '1',
    'show_last_items' => '1',
    'enable_pf_feature' => '1',
    'log_connections' => '0',
    'log_accessed' => '1',
    'time_format' => 'H:i:s',
    'date_format' => 'm/d/Y',
    'duplicate_folder' => '1',
    'item_duplicate_in_same_folder' => '1',
    'duplicate_item' => '1',
    'number_of_used_pw' => '6',
    'manager_edit' => '1',
    'cpassman_dir' => '/var/www/html',
    'cpassman_url' => 'http://172.16.1.112/',
    'favicon' => 'http://172.16.1.112/favicon.ico',
    'path_to_upload_folder' => '/var/www/html/upload',
    'url_to_upload_folder' => 'http://172.16.1.112/upload',
    'path_to_files_folder' => '/var/www/html/files',
    'url_to_files_folder' => 'http://172.16.1.112/files',
    'activate_expiration' => '0',
    'pw_life_duration' => '0',
    'maintenance_mode' => '0',
    'enable_sts' => '0',
    'encryptClientServer' => '1',
    'cpassman_version' => '2.1.27',
    'ldap_mode' => '0',
    'ldap_type' => '0',
    'ldap_suffix' => '0',
    'ldap_domain_dn' => '0',
    'ldap_domain_controler' => '0',
    'ldap_user_attribute' => '0',
    'ldap_ssl' => '0',
    'ldap_tls' => '0',
    'richtext' => '0',
    'allow_print' => '1',
    'roles_allowed_to_print' => '["["1","2"]"]',
    'show_description' => '1',
    'anyone_can_modify' => '0',
    'anyone_can_modify_bydefault' => '0',
    'nb_bad_authentication' => '0',
    'utf8_enabled' => '1',
    'restricted_to' => '0',
    'restricted_to_roles' => '0',
    'enable_send_email_on_user_login' => '0',
    'enable_user_can_create_folders' => '1',
    'insert_manual_entry_item_history' => '0',
    'enable_kb' => '0',
    'enable_email_notification_on_item_shown' => '0',
    'enable_email_notification_on_user_pw_change' => '0',
    'custom_logo' => '',
    'custom_login_text' => '',
    'default_language' => 'English',
    'send_stats' => '1',
    'get_tp_info' => '1',
    'send_mail_on_user_login' => '0',
    'nb_items_by_query' => 'auto',
    'enable_delete_after_consultation' => '0',
    'enable_personal_saltkey_cookie' => '0',
    'personal_saltkey_cookie_duration' => '0',
    'email_smtp_server' => '<omitted>',
    'email_smtp_auth' => '1',
    'email_auth_username' => '<omitted>',
    'email_auth_pwd' => '<omitted>',
    'email_port' => '587',
    'email_security' => 'tls',
    'email_server_url' => 'http://172.16.1.112',
    'email_from' => '<omitted>',
    'email_from_name' => 'Support',
    'pwd_maximum_length' => '40',
    'delay_item_edition' => '0',
    'allow_import' => '0',
    'proxy_ip' => '',
    'proxy_port' => '',
    'upload_maxfilesize' => '10mb',
    'upload_docext' => 'doc,docx,dotx,xls,xlsx,xltx,rtf,csv,txt,pdf,ppt,pptx,pot,dotx,xltx',
    'upload_imagesext' => 'jpg,jpeg,gif,png',
    'upload_pkgext' => '7z,rar,tar,zip',
    'upload_otherext' => 'sql,xml',
    'upload_imageresize_options' => '1',
    'upload_imageresize_width' => '800',
    'upload_imageresize_height' => '600',
    'upload_imageresize_quality' => '90',
    'use_md5_password_as_salt' => '0',
    'ga_website_name' => 'Clarity TeamPass',
    'api' => '0',
    'subfolder_rights_as_parent' => '1',
    'show_only_accessible_folders' => '0',
    'enable_suggestion' => '1',
    'otv_expiration_period' => '3',
    'default_session_expiration_time' => '60',
    'duo' => '0',
    'send_stats_time' => '0',
    'tree_counters' => '1',
    'item_extra_fields' => '1',
    'enable_attachment_encryption' => '1',
    'copy_to_clipboard_small_icons' => '1',
    'settings_offline_mode' => '0',
    'offline_key_level' => '0',
    'bck_script_filename' => 'bck_cpassman',
    'bck_script_path' => '/var/www/html//backups',
    'can_create_root_folder' => '0',
    'encryption_protocol' => 'ctr',
    'menu_type' => 'context',
    'google_authentication' => '0',
    'enable_server_password_change' => '0',
    'syslog_enable' => '0',
    'syslog_host' => 'localhost',
    'syslog_port' => '514',
    'saltkey_ante_2127' => '13',
    'teampass_version' => '3.1.1',
    'migration_to_2127' => 'done',
    'manager_move_item' => '0',
    'create_item_without_password' => '0',
    'agses_authentication_enabled' => '0',
    'timezone' => 'America/Chicago',
    'personal_saltkey_security_level' => '0',
    'ldap_new_user_is_administrated_by' => '0',
    'ldap_port' => '389',
    'enable_http_request_login' => '0',
    'otv_is_enabled' => '1',
    'ldap_and_local_authentication' => '0',
    'secure_display_image' => '1',
    'upload_zero_byte_file' => '0',
    'upload_all_extensions_file' => '0',
    'files_with_defuse' => 'done',
    'send_statistics_items' => 'stat_country;stat_users;stat_items;stat_items_shared;stat_folders;stat_folders_shared;stat_admins;stat_managers;stat_ro;stat_mysqlversion;stat_phpversion;stat_teampassversion;stat_languages;stat_kb;stat_suggestion;stat_customfields;stat_api;stat_2fa;stat_agses;stat_duo;stat_ldap;stat_syslog;stat_stricthttps;stat_fav;stat_pf;',
    'admin_2fa_required' => '1',
    'password_overview_delay' => '4',
    'roles_allowed_to_print_select' => '',
    'clipboard_life_duration' => '30',
    'mfa_for_roles' => '',
    'settings_tree_counters' => '0',
    'enable_massive_move_delete' => '0',
    'email_debug_level' => '0',
    'ga_reset_by_user' => '',
    'onthefly-backup-key' => '',
    'onthefly-restore-key' => '',
    'ldap_user_dn_attribute' => '',
    'ldap_dn_additional_user_dn' => '',
    'ldap_user_object_filter' => '',
    'ldap_bdn' => '',
    'ldap_hosts' => '',
    'ldap_password' => '',
    'ldap_username' => '',
    'api_token_duration' => '60',
    'enable_tasks_manager' => '1',
    'task_maximum_run_time' => '300',
    'maximum_number_of_items_to_treat' => '300',
    'tasks_manager_refreshing_period' => '100',
    'ldap_tls_certifacte_check' => 'LDAP_OPT_X_TLS_NEVER',
    'enable_tasks_log' => '0',
    'enable_ad_users_with_ad_groups' => '0',
    'enable_ad_user_auto_creation' => '0',
    'ldap_group_object_filter' => '',
    'ldap_guid_attibute' => 'objectguid',
    'sending_emails_job_frequency' => '2',
    'user_keys_job_frequency' => '1',
    'items_statistics_job_frequency' => '5',
    'reload_cache_table_task' => '',
    'rebuild_config_file' => '',
    'purge_temporary_files_task' => '',
    'clean_orphan_objects_task' => '',
    'users_personal_folder_task' => 'hourly;',
    'maximum_session_expiration_time' => '60',
    'items_ops_job_frequency' => '1',
    'upgrade_timestamp' => '1704822631',
    'enable_refresh_task_last_execution' => '1',
    'item_creation_templates' => '1',
    'ldap_group_objectclasses_attibute' => 'top,groupofuniquenames',
);

### Client configuration
**Browser:**
Tested current versions of Chrome, Safari, and Firefox

**Operating system:**
Windows 10, and macOS Sonoma
#### Web server error log
[Tue Jan 09 13:16:55.107922 2024] [proxy_fcgi:error] [pid 35791] [client 172.16.1.100:53184] AH01071: Got error 'PHP message: PHP Warning:  Undefined variable $query in /var/www/html/sources/users.datatable.php on line 187; PHP message: PHP Fatal error:  Uncaught Error: Call to a member function filter() on null in /var/www/html/sources/users.datatable.php:187\nStack trace:\n#0 {main}\n  thrown in /var/www/html/sources/users.datatable.php on line 187', referer: http://172.16.1.112/index.php?page=users
```
Insert your webserver log here
```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
GET http://172.16.1.112//sources/users.datatable.php?draw=1&columns%5B0&5D&5Bd jguery-min.js:2 G
aart=0&length=10&search&5Bvalue%5D=&search&5Bregex&5D=false& =1704831083556 500
100

send
@ iguery.min..js:2

ajax
@ iquery.min.js:2

@ iquery.dataTables.min.)s: 37

@ iquery.dataTables.min.js: 37

@ iquery.dataTables.min.js:29

@ iquery.dataTables.min.js:31

ha
@ iquery.dataTables.min.js:48

@ jguery.dataTables.min.)s:89

@ jguery.min.)s:2

fireWith
@ iquery.min.js:2

@ iquery.min.)s:2

(anonymous)
@ iquery.min.js:2

load (async)

send
@ iquery-min.js:2

ajax
@ iquery.min.js:2

(anonymous)
@ iquery.dataTables.min.js:88

each
@ iquery.min.js:2

each
@ jguery.min.js:2

@ jguery.dataTables.min.js:83

h. fn. DataTable @ jguery.dataTables.min.js: 165

(anonymous )
@ index.php?page=users: 3071
```
Insert the log here and especially the answer of the query that failed.
```

**Comment by deanbruhn** [comment]
I have run this test again, and the issue seems to be fixed in 3.1.1.31. Whole new issue now. 

[Fri Jan 12 12:32:20.871395 2024] [proxy_fcgi:error] [pid 33229] [client 172.16.1.105:51389] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught ArgumentCountError: setcookie(): Expects exactly 3 arguments when argument #3 ($expires_or_options) is an array in /var/www/html/sources/items.queries.php:3828\nStack trace:\n#0 /var/www/html/sources/items.queries.php(3828): setcookie()\n#1 {main}\n  thrown in /var/www/html/sources/items.queries.php on line 3828', referer: http://172.16.1.112/index.php?page=items

**Comment by sjsarg** [comment]
I got the same issue when I select the "User" option on the left menu. 
The users list does not appear and a box "Loading" appears

If a user is searched, there is no output

The issue appeared at version 3.1.1.27 and 3.1.1.41 . On version 3.0.6 it was working OK.
